File: Leetcode Solns/53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        s = 0
        maxx = nums[0]
        for i in nums:
            s += i
            maxx = max(s,maxx)
            if s<0:
                s = 0
        return maxx 

File: Leetcode Solns/53. Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = INT_MIN; 
        int currSum =0; 
        for(int val : nums){
            currSum += val;
            maxSum = max(currSum , maxSum);
            if (currSum < 0) currSum = 0;
        }    
        return maxSum;
    }
};
